### PR TITLE
Jb/singleselect downshift

### PR DESCRIPTION
**Pull request by Jbarget**
- creates `/components/Select/...`
- moves shared components into shared folder
- creates SingleSelect

### CODE REVIEW

File: reg-simple-ui/components/Select/SingleSelect/SingleSelect.stories.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    name: s,
+    id: s.toLowerCase()
+  }));
+
+const GridWrapper = styled.div`
+  width: 25%;
+`
+
+const SingleSelectTest = ({ placeholder, disabled, outline, icon }) => {
+  const [ value, setValue ] = useState({ name: '', id: '' })
```
Comment (mantagen) [inline reg-simple-ui/components/Select/SingleSelect/SingleSelect.stories.js]: Should default not be `null`?

File: reg-simple-ui/components/Select/shared/SelectController.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    props.isOpen &&
+    `
+    border-color: ${getColor('azure')};
+  `};
+  ${props =>
+    props.outline && outlineStyles};
+
+  &::placeholder {
+    color: ${getColor('graySuit')};
+  }
```
Comment (mantagen) [inline reg-simple-ui/components/Select/shared/SelectController.js]: Realised there's no `:focus` state too

File: reg-simple-ui/components/Select/SingleSelect/SingleSelect.js [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+import Downshift from 'downshift';
+
+import { Input, InputWrapper } from '../shared/SelectInput.js';
+import { InputButton } from '../shared/InputArrowIcon.js';
+import { Menu, MenuItem } from '../shared/DropdownMenu.js';
+import Controller from '../shared/SelectController.js';
+import Icon from '../../Icon/Icon.js';
+
+const shouldShowTick = ({ selectedItem, item }) => {
+  if (!selectedItem) return null
```
[the remark this answers is not in the document]
Comment (Jbarget) [reply reg-simple-ui/components/Select/SingleSelect/SingleSelect.js]: no need for what anymore?

File: reg-simple-ui/components/Icon/Tick.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+const Tick = props => (
+  <svg width='16px' height='16px' xmlns="http://www.w3.org/2000/svg" {...props}>
+    <path
+      d="M7.475 8.603l3.535-3.535.708.707-4.243 4.243-.707-.708L5 7.543l.707-.707 1.768 1.767z"
+     fillRule="evenodd"
+     fill={'currentColor'}
```
[the remark this answers is not in the document]
Comment (Jbarget) [reply reg-simple-ui/components/Icon/Tick.js]: will adjust this for consistency in another PR

File: reg-simple-ui/components/Select/SingleSelect/SingleSelect.stories.js (hunk 1 above)
Comment (Jbarget) [reply reg-simple-ui/components/Select/SingleSelect/SingleSelect.stories.js]: image

Comment (Jbarget) [reply reg-simple-ui/components/Select/SingleSelect/SingleSelect.stories.js]: when the value is null it thinks its an uncontrolled input

File: reg-simple-ui/components/Select/SingleSelect/SingleSelect.js [hunk 5]
```diff
@@ -0,0 +1,174 @@
+import React, { Component } from 'react';
+import Downshift from 'downshift';
+
+import { Input, InputWrapper } from '../shared/SelectInput.js';
+import { InputButton } from '../shared/InputArrowIcon.js';
+import { Menu, MenuItem } from '../shared/DropdownMenu.js';
+import Controller from '../shared/SelectController.js';
+import Icon from '../../Icon/Icon.js';
+
+const shouldShowTick = ({ selectedItem, item }) => {
```
[the remark this answers is not in the document]
Comment (Jbarget) [reply reg-simple-ui/components/Select/SingleSelect/SingleSelect.js]: changed it into a react functional component instead, its a pattern Fat Llama was using that i thought was quite nice

File: reg-simple-ui/components/Select/SingleSelect/SingleSelect.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (mantagen) [reply reg-simple-ui/components/Select/SingleSelect/SingleSelect.js]: no need to commit with `-n`

File: reg-simple-ui/components/Select/SingleSelect/SingleSelect.js [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+              disabled={disabled}
+              outline={outline}
+            >
+              <InputWrapper hasIcon={Boolean(icon)}>
+                {icon && <Icon icon={icon} width='12px' height='12px' color='waterloo' />}
+                <Input
+                  {...getInputProps({
+                    ref: this.input,
+                    readOnly: true,
+                    value: this.props.value ? this.props.value.name : null,
```
[the remark this answers is not in the document]
Comment (mantagen) [reply reg-simple-ui/components/Select/SingleSelect/SingleSelect.js]: That way the initial value can be `null` or `undefined`?

**Pull request merged**